libs/langchain-community/src/chat_models/llama_cpp.ts

```diff
@@ -87,20 +91,35 @@ export class ChatLlamaCpp extends SimpleChatModel<LlamaCppCallOptions> {
     return "ChatLlamaCpp";
   }
 
-  constructor(inputs: LlamaCppInputs) {
+  private constructor(inputs: LlamaCppInputs) {
```

Comment:
Does this make inheritance impossible? Should it be `protected` instead?

Reply:
And is there no sync API anymore?

Reply:
Unfortunately, there isn't a synchronous API anymore so the current implementation is the best workaround right now. This is both outlined in the [docs](https://node-llama-cpp.withcat.ai/guide/chat-session) and the [source code](https://github.com/withcatai/node-llama-cpp/blob/master/src/bindings/Llama.ts#L314). And yea it does make inheritance impossible, so will change to protected, as long as the person using the library is unable to access the constructor directly, all should be fine.